### PR TITLE
Add drive api

### PR DESCRIPTION
**Pull request by qunoissue**
This PR adds some functions to manipulate files on Google Drive.

Please see [here](https://github.com/qnoyxu/tonatona-google-server-api/tree/add-drive-api/sample) for a sample code.

### CODE REVIEW

**Comment by arowM** [comment]
Cool!
It seems CI fails, so could you check and fix it?

**Comment by qunoissue** [comment]
I've fixed package dependencies.

File: src/Google/Client.hs [hunk 1]
```diff
@@ -181,6 +226,47 @@ postGmailSend token email = do
...
+      (Just "multipart")
+      body)
+    (mkClientEnv manager googleBaseUrl)
+
+exportDriveFile ::
+     Response.Token
+  -> Text
+  -> Text
+  -> IO (Either ServantError BS.ByteString)
+exportDriveFile token fileId mimeType = do
```
Comment (arowM) [inline src/Google/Client.hs]: I would like to make `mimeType` and `fileId` type safe.
Is it possible to create new type `MimeType` and `FileId`?

The `query` and `orderBy` used elsewhere should also be type safe, but if it's too labor intensive, you can just leave it.

**Review by arowM — CHANGES_REQUESTED** [review]

**Comment by qunoissue** [comment]
Commit [cdc5f8b](https://github.com/qnoyxu/haskell-google-server-api/commit/cdc5f8b80137c08f40b2a2825e9d826b39a73c38) makes some types safe.
I created new type of `query` by simply wrapping `Text` because it seems to be used only here and its syntax is complicated.

Commit [914af55](https://github.com/qnoyxu/haskell-google-server-api/commit/914af55fd6aa86ff7f66984ed9dc2985c589e174) renames `exportDriveFile` to `downloadDriveFile`.

**Comment by arowM** [comment]
I apologize for not being explicit, and for going through your previous PR, but the `Form` module is actually supposed to export a chunk of data that you want to create such as `Form.CalendarEvent`.

**Comment by qunoissue** [comment]
I'm not sure I understand what 'chunk' means, but does it mean that not only the type that represents `contentType`, but also types that have a single field (such as `FileId`), should be exported from another file?

**Comment by arowM** [comment]
Yeah, it's what I meant.
There are two ways to design `downloadDriveFIle` API:
a. Take token and two arguments fileId and mimeType exported from some modules such as `Type` you made before
b. Take token and one argument exported from `Form`
    ```
    { fileId :: Text
    , mimeType :: MimeType
    }
    ```

**Comment by qunoissue** [comment]
Thanks for the great suggestions!
I changed type definitions accoding to the proposal b.
The types which were exported from `Form` and `Response` respectively such as `FileId` are now commonized.

**Comment by arowM** [comment]
Great!

Could you create a new PR adding your sample directory?
It's helpfull.

**Pull request merged**